support/cas-server-support-thymeleaf/src/main/resources/templates/fragments/theme.html

```diff
@@ -0,0 +1,28 @@
+<span th:fragment="javascript" th:remove="tag">
+    <span th:remove="tag" th:if="${'material' == #strings.defaultString(#themes.code('cas.theme.framework'), 'material')}">
```

Comment:
This cannot scale. Themes must be self-contained. If you add theme XYZ tomorrow that means this block here must be modified as well. If I remove the XYZ theme 2 years from now, I'd have to remember to come here and also remove the bit from the if block. This cannot help with maintenance and can only lead to more work. I am against more work :)


What you should do instead is,

1. Remove the framework type setting. A setting must be modifiable by the user when made public. This is just used an internal condition. So it cannot be a setting. 
2. A theme should just be able to override this fragment here with the appropriate includes/excludes. The default would be material, and a theme could override this bit to include its own. By removing the condition, you rely in polymorphic behavior essentially to have overrides replace defaults.

Reply:
You should also rename the fragment to not be `theme.html`. Name it after the responsibilities of the component, like "includes.html" or something like that.